Uploader/Program.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using NLog;
+using System.IO;
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading;
+using System.Timers;
+using Newtonsoft.Json;
+using System.IO.Abstractions;
+using System.Collections.Generic;
+
+/// <summary>
+/// Main entry point of the program.
+/// </summary>
+public class Uploader
+{
+    private const long TimerIntervalInMilliseconds = 1000 * 60 * 5;
+    private static readonly Logger logger = LogManager.GetCurrentClassLogger();
+
+    private readonly string traceDirectory;
+    private readonly Config config;
+    private readonly TimerAction timerAction;
+
+    /// <summary>
+    /// Main entry point. Expects a single argument: the path to a directory that contains the trace files to upload.
+    /// </summary>
+    public static void Main(string[] args)
+    {
+        if (IsAlreadyRunning())
+        {
+            Console.WriteLine("Another instance is already running.");
+            return;
```

Comment:
return error code?

Reply:
No since this is not an error, this is what's supposed to happen

Reply:
true...